rfcs/text/0000-data_stream-fields.md

```diff
@@ -0,0 +1,137 @@
+# 0000: Data stream fields
+<!-- Leave this ID at 0000. The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC. -->
+
+- Stage: **1 (proposal)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
+- Date: **TBD** <!-- The ECS team sets this date at merge time. This is the date of the latest stage advancement. -->
+
+When introducing the new indexing strategy for Elastic Agent which uses data streams, we found that adding a few "constant_keyword" fields corresponding to the central components in the new indexing strategy would be advantagous. 
+
+
+<!--
+As you work on your RFC, use the "Stage N" comments to guide you in what you should focus on, for the stage you're targeting.
+Feel free to remove these comments as you go along.
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+## Fields
+
+<!--
+Stage 1: Describe at a high level how this change affects fields. Which fieldsets will be impacted? How many fields overall? Are we primarily adding fields, removing fields, or changing existing fields? The goal here is to understand the fundamental technical implications and likely extent of these changes. ~2-5 sentences.
+-->
+
+This RFC proposes to introduce a new fieldset called "data_stream". The fieldset consists of the following fields:
+Field     | Mapping type | Description
+----------|--------------|--------------
+data_stream.type | constant_keyword | An overarching type for the data stream. Currently allowed values include "logs", "metrics". We expect to also add "traces" and "synthetics" in the near future
+data_stream.dataset | constant_keyword | A copy of event.dataset
```

Comment:
Suggestion: Lets mention here that if not value for it is set, this should fall back to `generic`.